Realm.Shared/Dynamic/DynamicRealmObjectHelper.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using Realms.Weaving;
+
+namespace Realms.Dynamic
+{
+    internal class DynamicRealmObjectHelper : IRealmObjectHelper
```

Comment:
I suppose that'd work just as well, although recently I've begun favoring fields instead of auto properties for pure storage, especially when it's internal to the assembly.

@kristiansupont @AndyDentFree, what do you think?